pkg/deviceplugin/plugin_test.go

```diff
@@ -106,54 +97,145 @@ var _ = Describe("Macvtap device plugin", func() {
 		cleanup()
 	})
 
-	It("should allocate a new device upon request", func() {
-		ifaceName := masterIfaceName + "Mvp99"
-		req := &pluginapi.AllocateRequest{
-			ContainerRequests: []*pluginapi.ContainerAllocateRequest{
-				{
-					DevicesIDs: []string{
-						ifaceName,
+	Describe("plugin", func() {
+		var mvdp dpm.PluginInterface
+		var sendSpy *ListAndWatchServerSendSpy
+
+		BeforeEach(func() {
+			mvdp = NewMacvtapDevicePlugin(masterIfaceName, masterIfaceName, "bridge", 0, testNs.Path())
+			sendSpy = &ListAndWatchServerSendSpy{}
+			go func() {
+				err := mvdp.ListAndWatch(nil, sendSpy)
+				Expect(err).NotTo(HaveOccurred())
+			}()
+		})
+
+		AfterEach(func() {
+			mvdp.(dpm.PluginInterfaceStop).Stop()
+		})
+
+		It("SHOULD allocate a new device upon request", func() {
+			ifaceName := masterIfaceName + "Mvp99"
+			req := &pluginapi.AllocateRequest{
+				ContainerRequests: []*pluginapi.ContainerAllocateRequest{
+					{
+						DevicesIDs: []string{
+							ifaceName,
+						},
 					},
 				},
-			},
-		}
+			}
+
+			// Allocate on its own goroutine as it would from the grpc server
+			// handler
+			var res *pluginapi.AllocateResponse
```

Comment:
This is distracting.

Could you try to encapsulate this on a function so it gives some insight as to what happens ?

I feel such low level stuff shouldn't belong in the test body - the test should be readable; this section just distracts me from what's being tested. Afraid I have no alternatives for you at the moment.

Reply:
Gone.